### PR TITLE
Validate RuntimeAttributes before starting a call

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by mcovarr** [comment]
also @cjllanwarne

File: src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala [hunk 1]
```diff
@@ -39,8 +39,10 @@ object CromwellRuntimeAttributes {
     validatedRuntimeAttributes match {
       case Success(r) => r
       case Failure(f) =>
-        val errorMessages = f.toList.mkString(", ")
-        throw new IllegalArgumentException(s"RuntimeAttribute is not valid: Errors: $errorMessages")
+        throw new IllegalArgumentException() with ThrowableWithErrors {
+          val message = "RuntimeAttribute is not valid."
+          val errors = f
+        }
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: will the resulting error message provide the context of the offending task?

Comment (Horneth) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: Yes, `ThrowableWithErrors` override `getMessage` to include the `errors`

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 2]
```diff
@@ -1061,7 +1061,7 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
         self ! InitialStartCall(callKey, CallActor.Start)
     }
 
-    if (backendCall.workflowDescriptor.readFromCache) {
+    def startCall = if (backendCall.workflowDescriptor.readFromCache) {
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: This is one `if` expression, but I think our standard is to have braces around a multiline function and this is a lot of lines.

File: src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala (hunk 1 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: and the name of the task will be in there?

**Comment by mcovarr** [comment]
:+1: with some nitpicks

File: src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala (hunk 1 above)
Comment (Horneth) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: Hmm the name of the task is in the logger prefix but not in the message, I can add it there too.
```
2016-01-15 11:59:27,542 cromwell-system-akka.actor.default-dispatcher-8 ERROR - WorkflowActor [UUID(978b16e7):hello]: RuntimeAttribute is not valid.
Missing required keys in runtime configuration for backend 'JES': docker
```

Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala]: OK, that's fine if it's in the prefix, I just wanted to make it went out with the message.  :smile:

**Comment by cjllanwarne** [comment]
:+1: after Miguel's comments

**Pull request merged**